src/content/docs/en/reference/api-reference.mdx

```diff
@@ -192,34 +125,145 @@ if (!post) {
 
 See also: [`params`](#params)
 
-### `Astro.request`
+## `Astro.url`
 
 <p>
 
-**Type:** `Request`
+**Type:** `URL`<br />
+<Since v="1.0.0" />
+</p>
+
+`url` is a [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) object constructed from the current `request.url` URL string value. Useful for interacting with individual properties of the request URL, like pathname and origin. 
+
+Equivalent to doing `new URL(Astro.request.url)`.
+
+`url` will be `localhost` in dev mode if [site](/en/reference/configuration-reference/#site) is not configured for static sites, and for on-demand rendered sites using `server` or `hybrid` output.
```

Comment:
```suggestion
`url` will be `localhost` in dev mode if [`site`](/en/reference/configuration-reference/#site) is not configured for static sites, and for on-demand rendered sites using `server` or `hybrid` output.
```

I think this would be more readable if `site` is in a inline code block, like it's done below:

```
Astro.site` returns a `URL` made from `site` in your Astro config. If `site` in your Astro config isn't defined, `Astro.site` won't be defined.
```